src/transport/PairingSession.h

```diff
@@ -50,11 +50,18 @@ class DLL_EXPORT PairingSession
 
     CATValues GetPeerCATs() const { return mPeerCATs; }
 
-    // TODO: the local key id should be allocateed at start
-    // mLocalSessionId should be const and assigned at the construction, such that GetLocalSessionId will always return a valid key
-    // id , and SetLocalSessionId is not necessary.
-    uint16_t GetLocalSessionId() const { return mLocalSessionId; }
-    bool IsValidLocalSessionId() const { return mLocalSessionId != kInvalidKeyId; }
+    Optional<uint16_t> GetLocalSessionId() const
+    {
+        Optional<uint16_t> localSessionId;
+        VerifyOrExit(mSecureSessionHolder, localSessionId = Optional<uint16_t>::Missing());
```

Comment:
```suggestion
        VerifyOrExit(mSecureSessionHolder, localSessionId = NullOptional);
```
is shorter...  but also, it's set to that already, by the Optional constructor.  Either way, though.